src/lib/openapi/spec/playground-constraint-schema.ts

```diff
@@ -0,0 +1,20 @@
+import { constraintSchemaBase } from './constraint-schema';
+import { FromSchema } from 'json-schema-to-ts';
+
+export const playgroundConstraintSchema = {
+    $id: '#/components/schemas/playgroundConstraintSchema',
+    additionalProperties: false,
+    ...constraintSchemaBase,
+    required: [...constraintSchemaBase.required, 'result'],
+    properties: {
+        ...constraintSchemaBase.properties,
+        result: {
+            description: 'Whether this was evaluated as true or false.',
+            type: 'boolean',
+        },
```

Comment:
I haven't followed this PR so feel free to ignore, but extending the constraint schema seems complicated, both here and on the frontend. Maybe we could add stuff next to the constraints and segments instead of extending?

```ts
{
    constraints: [...], // Regular constraints.
    segments: [...], // Regular segments.
    results: [
    	{
    		constraintIndex: 1, // A constraint ID would be nice.
    		enabled: true,
    	},
    	{
    		constraintIndex: 1,
    		segmentId: 2,
    		enabled: true,
    	}
    	// ...
    ]
}
```

Reply:
Hmm, yes and no? We could definitely do that, but I'm not sure I agree. This might just be a personal thing, but it strikes me as natural to put the result right next to what it's evaluating. But you do make a good point that it does add more data models. Without it, we could potentially reuse some of the types from before. That said, from an API consumer view, I think I'd prefer to have the result colocated 🤷🏼

Reply:
Yeah, makes sense. I do want to see if we can avoid the `constraint: IConstraint | PlaygroundConstraintSchema` stuff on the frontend though. Would be nice if the playground-related types were kept there.

Reply:
Oh, that's interesting. I haven't seen the frontend implementation, so I don't know. Why do you need that? Couldn't you just use IConstraint? A playgroundconstraintschema should fit into an IConstraint, doesn't it?

Reply:
Not sure, haven't looked at it in detail. Might just be because it's `IConstraint` and not `ConstraintSchema`. :shrug: